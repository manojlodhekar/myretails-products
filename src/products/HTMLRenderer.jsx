import React from 'react';

const HTMLRenderer = (props) =>{
    return(
        <div>
            <div style={{paddingTop: 10}}>
                <span id="htmlRendererId" className="returnPolicyCls" dangerouslySetInnerHTML={{__html: props.htmlContents}} />
            </div>
        </div>
    )
}
export default HTMLRenderer;