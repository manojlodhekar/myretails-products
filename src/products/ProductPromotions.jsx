import React from 'react';
import { Icon, Divider } from 'semantic-ui-react';

const ProductPromotions = (props) =>{

    const renderPromotionDetails = () =>{

        let productDetail =  props.promotionDetails.map((item) => {
            return(
                <div className="promotions" key={item.promotionIdentifier}>
                    <div>
                        <Icon name='tag'/>{item.Description[0].shortDescription} 
                    </div>
                </div>
            )
        });
        return productDetail;
    }

    return(
        <div>
            <Divider />
            {renderPromotionDetails()}
            <Divider />
        </div>
    )
}
export default ProductPromotions;