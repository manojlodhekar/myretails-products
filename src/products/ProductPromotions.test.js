import React from 'react';
import ReactDOM from 'react-dom';
import ProductPromotions from './ProductPromotions';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import HTMLRenderer from './HTMLRenderer';

jest.mock('react-dom');

Enzyme.configure({ adapter: new Adapter() });
let productPromotions;
const itemPayload =  {"Promotions": [
  {
   "Description": [
    {
     "legalDisclaimer": "Offer available online only. Offer applies to purchases of $50 or more of eligible items across all categories. Look for the &quot;SPEND $50: SHIPS FREE&quot; logo on eligible items. Some exclusions apply. Items that are not eligible are subject to shipping charges. $50 purchase is based on eligible merchandise subtotal. Items that are not eligible, GiftCards, e-GiftCards, gift wrap, tax and shipping and handling charges will not be included in determining merchandise subtotal. Offer valid for orders shipping within the 48 contiguous states, as well as APO\/FPO and for Standard and To the Door shipping methods only. Not valid on previous orders.",
     "shortDescription": "SPEND $50, GET FREE SHIPPING"
    }
   ],
   "endDate": "2014-05-25 06:59:00.001",
   "promotionIdentifier": "10736506",
   "promotionType": "Buy catalog entries from category X, get shipping at a fixed price",
   "startDate": "2014-05-18 07:00:00.001"
  }]};
beforeEach(()=>{
  productPromotions = mount(<ProductPromotions promotionDetails={itemPayload}>
  </ProductPromotions>);

})
describe('ProductPromotions is defined', () => {

  it('should render correctly', () => {
    expect(productPromotions).toHaveLength(1);
  });

  it('should render inner component', () => {
    expect(productPromotions).toMatchSnapshot();
  });

  it('should render curent productPromotions of component', () => {
    productPromotions = shallow(<ProductPromotions promotionDetails={itemPayload.Promotions} />);
    expect(productPromotions.find('.promotions').first().text()).toEqual('<Icon />SPEND $50, GET FREE SHIPPING');
  });
  it('should render dscription product Promotions of component', () => {
    productPromotions = shallow(<ProductPromotions promotionDetails={itemPayload.Promotions} />);
    expect(productPromotions.find('.promotions')).toHaveLength(1);
  });


 });
