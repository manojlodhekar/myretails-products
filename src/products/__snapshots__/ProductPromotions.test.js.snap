// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPromotions is defined should render inner component 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <ProductPromotions
    promotionDetails={
        Object {
            "Promotions": Array [
              Object {
                "Description": Array [
                  Object {
                    "legalDisclaimer": "Offer available online only. Offer applies to purchases of $50 or more of eligible items across all categories. Look for the &quot;SPEND $50: SHIPS FREE&quot; logo on eligible items. Some exclusions apply. Items that are not eligible are subject to shipping charges. $50 purchase is based on eligible merchandise subtotal. Items that are not eligible, GiftCards, e-GiftCards, gift wrap, tax and shipping and handling charges will not be included in determining merchandise subtotal. Offer valid for orders shipping within the 48 contiguous states, as well as APO/FPO and for Standard and To the Door shipping methods only. Not valid on previous orders.",
                    "shortDescription": "SPEND $50, GET FREE SHIPPING",
                  },
                ],
                "endDate": "2014-05-25 06:59:00.001",
                "promotionIdentifier": "10736506",
                "promotionType": "Buy catalog entries from category X, get shipping at a fixed price",
                "startDate": "2014-05-18 07:00:00.001",
              },
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
